library/core/src/num/nonzero.rs

```diff
@@ -111,6 +111,12 @@ impl_zeroable_primitive!(
 /// ```
 ///
 /// [null pointer optimization]: crate::option#representation
```

Comment:
shouldn't this be below any textual comments?

Reply:
CommonMark actually doesn't impose any ordering requirement onto link reference definitions, they can even appear before the link that references them, however, conventional style is to put them at the end of the docs or section.

I think the real issue here is that you've technically added this note into the "Layout" section despite it not having anything to do with layout.  To fix this (as well as the concern about link reference definition placement), either move the note before the first heading, or add a heading like `# Note on generic usage` before the note.